src/pytorch_lightning/utilities/parsing.py

```diff
@@ -179,8 +180,8 @@ def collect_init_args(
     if not isinstance(frame.f_back, types.FrameType):
         return path_args
 
-    if "__class__" in local_vars and (not classes or issubclass(local_vars["__class__"], classes)):
-        local_args = get_init_args(frame)
+    local_self, local_args = get_init_args(frame)
```

Comment:
```suggestion
    local_vars, local_args = get_init_args(frame)
```
is this a typo?

Reply:
No, why? `local_self` is the value of `self` (or whatever the first argument is called) in the current frame.